cxx/distributions/normal_test.cc

```diff
@@ -4,18 +4,65 @@
 
 #include "distributions/normal.hh"
 
+#include <boost/math/distributions/inverse_gamma.hpp>
+#include <boost/math/distributions/normal.hpp>
 #include <boost/test/included/unit_test.hpp>
+
+#include "util_math.hh"
 namespace tt = boost::test_tools;
 
+BOOST_AUTO_TEST_CASE(test_against_boost) {
+  // Construct a biased estimator of the log_prob using Boost, and check that we
+  // are within tolerance.
+  std::mt19937 prng;
+  std::mt19937 prng2;
+  Normal nd(&prng);
+
+  std::vector<double> inverse_gamma_samples;
+  std::vector<double> normal_samples;
+  int num_trials = 5000;
+
+  for (int j = 0; j < num_trials; ++j) {
+    double x = 1. / (std::gamma_distribution<double>(nd.v, nd.s))(prng2);
+    inverse_gamma_samples.emplace_back(x);
+    double y = (std::normal_distribution<double>(nd.m, sqrt(x / nd.r)))(prng2);
+    normal_samples.emplace_back(y);
+  }
+
+  double accumulated_log_prob;
+  boost::math::inverse_gamma_distribution<> inv_gamma_dist(nd.v, nd.s);
+
+  for (int i = 0; i < 10; ++i) {
+    nd.incorporate(i);
+    std::vector<double> average_log_prob;
+
+    for (int j = 0; j < num_trials; ++j) {
+      double ig = inverse_gamma_samples[j];
+      double n = normal_samples[j];
+      boost::math::normal_distribution normal_dist(n, sqrt(ig));
+      average_log_prob.emplace_back(log(
+          boost::math::pdf(normal_dist, static_cast<double>(i)) / num_trials));
+    }
+    accumulated_log_prob += logsumexp(average_log_prob);
+    BOOST_TEST(nd.logp_score() == accumulated_log_prob, tt::tolerance(0.3));
```

Comment:
It's lower for earlier iterations, but not universally (since the magnitude of the log_prob changes). Part of the problem is this is a biased estimator. I'm going to revisit this code, since the right thing would be doing quadrature to approximate the prob function, but I'm not too familiar with the quadrature code in boost so haven't done that yet. (Same with beta_bernoulli).